bundles/binding/org.openhab.binding.rwesmarthome/src/main/java/org/openhab/binding/rwesmarthome/internal/communicator/RWESmarthomeSession.java

```diff
@@ -0,0 +1,663 @@
+	}
+
+	/**
+	 * Refresh logical device state.
+	 * 
+	 * @return the string
+	 * @throws SmartHomeSessionExpiredException
+	 *             the smart home session expired exception
+	 */
+	public String refreshLogicalDeviceStates() throws RWESmarthomeSessionExpiredException {
+
+		String getAllLogicalDeviceStatesRequest;
+		String sResponse = "";
+		getAllLogicalDeviceStatesRequest = "<BaseRequest xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:type=\"GetAllLogicalDeviceStatesRequest\" Version=\"" + FIRMWARE_VERSION + "\" RequestId=\""
+				+ requestId
+				+ "\" SessionId=\""
+				+ this.getSessionId()
+				+ "\" BasedOnConfigVersion=\""
+				+ currentConfigurationVersion
+				+ "\" />";
+		sResponse = executeRequest(getAllLogicalDeviceStatesRequest, "/cmd");
+
+		try {
+			GetAllLogicalDeviceStatesXMLResponse entitiesXMLRes = new GetAllLogicalDeviceStatesXMLResponse(IOUtils.toInputStream(sResponse, "UTF8"));
+
+//			this.currentConfigurationVersion = entitiesXMLRes.getConfigurationVersion();
+		} catch (IOException e) {
+			throw new RWESmarthomeSessionExpiredException(e);
+		}
+		
+		logger.debug("Refresh LD Response: " + sResponse);
+		return sResponse;
+	}
+
+	/**
+	 * Gets the notifications.
+	 * @return 
+	 * 
+	 * @return the XML response
+	 * @throws SmartHomeSessionExpiredException
+	 *             the smart home session expired exception
+	 * @throws LogoutNotificationException, SmartHomeSessionExpiredException
+	 * @throws ConfigurationChangedException 
+	 */	
+	public String getNotifications() throws LogoutNotificationException, RWESmarthomeSessionExpiredException, ConfigurationChangedException {
+			
+			String sResponse = getNotificationsRequest();
+
+			// check for logout first			
+			if(sResponse.contains("LogoutNotification") || sResponse.contains("ConfigurationChangedNotification")) {
+				// logout from notification updates received
+				NotificationsXMLResponse notXmlRes = new NotificationsXMLResponse(IOUtils.toInputStream(sResponse));
+			}
+			
+			// as the XML format is almost the same, we can reuse GetAllLogicalDeviceStatesXMLResponse
+			// for the LogicalDevicesStatesChangedNotification
+			GetAllLogicalDeviceStatesXMLResponse logDevXmlRes = new GetAllLogicalDeviceStatesXMLResponse(IOUtils.toInputStream(sResponse));
+			
+			logger.debug("Notifications: {}", sResponse);
+			
+			return sResponse;
+	}
+	
+	
+	/**
+	 * Sends the getNotificationsRequest.
+	 * 
+	 * @return the XML response
+	 * @throws RWESmarthomeSessionExpiredException
+	 */
+	public String getNotificationsRequest() throws RWESmarthomeSessionExpiredException {
+		return executeRequest("upd", "/upd");
+	}
+
+	/**
+	 * Switch actuator change state.
+	 * 
+	 * @param deviceId
+	 *            the device id
+	 * @param on
+	 *            the new switchstate
+	 * @throws RWESmarthomeSessionExpiredException
+	 *             the smart home session expired exception
+	 */
+	public void switchActuatorChangeState(String deviceId, boolean on)
```

Comment:
i wouldn't expect a Session to contain code to switch something. A session sounds like a context object and thus should better be a parameter to be delivered to a method located in the Communitcator (see also subsequent action methods).